docs/reference/hyperparameter_tuner.md

```diff
@@ -0,0 +1,196 @@
+  tuning cannot be performed and an error is raised.
+- **n_trials**: *int, default=None*  
+  "The number of trials. If this argument is set to None, there is no limitation on the number of trials.
+  If timeout is also set to None, the study continues to create trials until it receives a termination signal such as
+  Ctrl+C or SIGTERM", taken from [here](https://optuna.readthedocs.io/en/stable/reference/generated/optuna.study.Study.html#optuna.study.Study.optimize).
+- **timeout**: *float, default=None*  
+  "Stop study after the given number of second(s).
+  If this argument is set to None, the study is executed without time limitation.
+  If n_trials is also set to None, the study continues to create trials until it receives a termination signal such as
+  Ctrl+C or SIGTERM", taken from [here](https://optuna.readthedocs.io/en/stable/reference/generated/optuna.study.Study.html#optuna.study.Study.optimize).
+- **n_jobs**: *int, default=None*  
+  "The number of parallel jobs. If this argument is set to -1, the number is set to CPU count", taken from [here](https://optuna.readthedocs.io/en/stable/reference/generated/optuna.study.Study.html#optuna.study.Study.optimize).
+- **show_progress_bar**: *bool, default=False*  
+  "Flag to show progress bars or not. To disable progress bar, set this False.
+  Currently, progress bar is experimental feature and disabled when n_jobs ≠1", taken from [here](https://optuna.readthedocs.io/en/stable/reference/generated/optuna.study.Study.html#optuna.study.Study.optimize).
+- **verbose**: *bool, default=False*  
+  If *True*, maximum verbosity is enabled.
+
+#### Demos and tutorial
+
+A demo showcasing the usage and functionality of the *HyperparameterTuner* is available
+[here](../../projects/utils/hyperparameter_tuner/hyperparameter_tuner_demo.py).
+Also, a tutorial in the form of a Jupyter Notebook is available
+[here](../../projects/utils/hyperparameter_tuner/hyperparameter_tuning_tutorial.ipynb).
+
+
+#### Examples
+
+* **Hyperparameter tuning example with the [DetrLearner](detr.md).**
+
+  This example shows how to tune hyperparameters of the *DetrLearner*.
+
+  ```python
+  from opendr.utils.hyperparameter_tuner.hyperparameter_tuner import HyperparameterTuner
+  from opendr.perception.object_detection_2d.detr.detr_learner import DetrLearner
+  from opendr.engine.datasets import ExternalDataset
+
+  # Create a coco dataset, containing training and evaluation data
+  dataset = ExternalDataset(path='./my_dataset', dataset_type='COCO')
+
+  # Specify the arguments that are required for the fit method
+  fit_arguments = {'dataset': dataset}
+
+  # Specify the arguments that are required for the eval method
+  eval_arguments = {'dataset': dataset}
+
+  # Specify timeout such that optimization is performed for 4 hours
+  timeout = 14400
+
+  # Initialize the tuner
+  tuner = HyperparameterTuner(DetrLearner)
+
+  # Optimize
+  best_parameters = tuner.optimize(
```

Comment:
There are no hyperparameters provided in this example. What is the behavior of the algorithm in this case?

Reply:
In that case, the hyperparameter tuner checks whether the get_hyperparameters method is implemented in the specified learner. This method can provide default values for the hyperparameter specifications. Implementing such a method in learners would improve the usability of the hyperparameter tuner I believe.